react/src/reactTextMask.js

```diff
@@ -42,15 +42,14 @@ export default class MaskedInput extends React.Component {
     delete props.onChange
     delete props.showMask
 
-    return (
-      <input
-        {...props}
-        onBlur={this.onBlur}
-        onChange={this.onChange}
-        defaultValue={this.props.value}
-        ref={(inputElement) => (this.inputElement = inputElement)}
-      />
-    )
+    const ref = (inputElement) => (this.inputElement = inputElement)
+
+    return render(ref, {
```

Comment:
Have you considered passing `ref` as part of the `props` object? 
I would think that it would make for a more intuitive API if all the necessary  attributes were to be pulled out from the `props` object instead of using `ref` as a separate argument. 

This way, your example from above would turn in something like this: 
```jsx
render={(props) => (
  <MyStyledInput innerRef={props.ref} {...props} />
)}
```

Reply:
I had it this way originally actually, but found that it didn't work as expected.  It seems that React strips out refs: https://stackoverflow.com/a/41790768

I agree that this is a cleaner and more intuitive API, but was unable to make it work.  I also considered making `ref` the second parameter, but it will always be needed and it might get left off it is seen as optional.

Reply:
Do  you think have `ref` set to a default value of `null` would make sense? I find that I won't be using `ref` a lot and having it set to `null` would allow users to not have to enter a value if it doesn't make sense to do so.

Reply:
It is my understanding that `ref` is necessary, as that's how the `core` libs get attached to the React-rendered `input` (https://github.com/text-mask/text-mask/blob/master/react/src/reactTextMask.js#L18).

However, the `render` prop itself is optional, so you shouldn't have to worry about `ref` unless you are trying to customize how your `input` gets rendered.  In that case, you'd need to point your rendered `input` to the `ref` passed in the render function.